### PR TITLE
Re-add castShadow property to light components and default to false

### PR DESCRIPTION
**Pull request by dinomut1**
For performance reasons we want to have lights not cast shadows by default

### CODE REVIEW

File: packages/engine/src/scene/components/PointLightComponent.ts [hunk 1]
```diff
@@ -11,6 +11,19 @@ import { ObjectLayers } from '../constants/ObjectLayers'
 import { setObjectLayers } from '../functions/setObjectLayers'
 import { addObjectToGroup, removeObjectFromGroup } from './GroupComponent'
 
+export type PointLightComponentType = {
```
Comment (HexaField) [inline packages/engine/src/scene/components/PointLightComponent.ts:14]: We can use ComponentType<typeof PointLightComponent> for these types

**Review by HexaField — CHANGES_REQUESTED** [review]
Is there a reason we can't use a ShadowComponent with cast set to false? That was the intention of removing the property previously

File: packages/engine/src/scene/components/PointLightComponent.ts (hunk 1 above)
Comment (dinomut1) [reply packages/engine/src/scene/components/PointLightComponent.ts]: Why not put the castShadow property with all of the other shadow properties that are in the light components?

Comment (dinomut1) [reply packages/engine/src/scene/components/PointLightComponent.ts]: Also, the lights need different default values for castShadow than meshes and whatnot (lights false, meshes true)

Comment (HexaField) [reply packages/engine/src/scene/components/PointLightComponent.ts]: > Why not put the castShadow property with all of the other shadow properties that are in the light components?

That's a good point, in this case we should remove remove castShadow from ShadowComponent and turn it into a tag component

**Review by speigg — APPROVED** [review]

File: packages/engine/src/scene/components/PointLightComponent.ts (hunk 1 above)
Comment (speigg) [reply packages/engine/src/scene/components/PointLightComponent.ts]: @HexaField what about receiving shadows? You want a tag component for casting and receiving each?

Comment (HexaField) [reply packages/engine/src/scene/components/PointLightComponent.ts]: Sorry, yes, I mean put castShadow on light components, and have receiveShadow be set only via the presence of a tag ShadowComponent

**Pull request merged**